cmd/snap-gdb-shim/snap-gdb-shim.c

```diff
@@ -15,20 +15,64 @@
  *
  */
 
+#include <errno.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
 #include "../libsnap-confine-private/utils.h"
 
+int must_parse_int(const char *s)
+{
+	char *endptr = NULL;
+
+	errno = 0;
+	long i = strtol(s, &endptr, 10);
+	if (errno != 0 || s == endptr || *endptr != '\0' || i < 0) {
```

Comment:
Given that we have https://github.com/snapcore/snapd/blob/379f29abd21494b24a35e3234692a59df9a2cef4/cmd/snap-confine/seccomp-support.c#L130 perhaps this could be extracted into one of the utilities and used in both places?